### PR TITLE
Include similarity in index settings (#2924)

### LINKED ISSUE #2890: Include Classic similarity in index settings
`CreateIndex()` allows a similarity to be set

```c#
client.CreateIndex("index", c => c
	.Similarity(s => s
		.BM25("default", ds => ds)
	)
);
```

but there appear to be two issues with this

1. No way to set the [Classic Similarity](https://www.elastic.co/guide/en/elasticsearch/reference/current/index-modules-similarity.html#classic-similarity) (looks like it is still referred to as Default in source)
2. Similarity settings should be under `settings.index` e.g.

```json
PUT /my_index
{
  "settings": {
    "index": {
      "similarity": {
        "default": {
          "type": "classic"
        }
      }
    }
  }
}
```

but NEST currently produces

```json
PUT http://localhost:9200/messages?pretty=true 
{
  "similarity": {
    "default": {
      "type": "BM25"
    }
  }
}
```

### PR DESCRIPTION
**Pull request by russcam**
Non binary breaking back port of 648c3cf9d016839bc6aa4dbd60da28d90db984b3

Similarity settings are nested under settings>index
Add ClassicSimilarity
Add DFISimilarity
Add similarity settings to create index integration tests
Add XML comments for each similarity
Refactor SimilarityJsonConverter to not look up the NEST ISimilarity type with Type.GetType() but to infer it from the type property value directly.
Add Similarity to the updatable index settings

Closes #2890

(cherry picked from commit bf78582ab961895ac8d2ac0e9c42696586a12aa4)

### CODE REVIEW

File: src/Nest/IndexModules/Similarity/DefaultSimilarity.cs [hunk 1]
```diff
@@ -2,26 +2,36 @@
...
+	/// <inheritdoc />
+	public class DefaultSimilarityDescriptor
 		: DescriptorBase<DefaultSimilarityDescriptor, IDefaultSimilarity>, IDefaultSimilarity
 	{
 		string ISimilarity.Type => "default";
 		bool? IDefaultSimilarity.DiscountOverlaps { get; set; }
 
-		public DefaultSimilarityDescriptor DiscountOverlaps(bool? discount = true) => Assign(a => a.DiscountOverlaps = discount);
+		/// <inheritdoc />
+		public DefaultSimilarityDescriptor DiscountOverlaps(bool discount = true) => Assign(a => a.DiscountOverlaps = discount);
```
Comment (russcam) [inline src/Nest/IndexModules/Similarity/DefaultSimilarity.cs]: Needs to be changed to `bool?` to maintain binary compatibility

File: src/Nest/IndexModules/Similarity/BM25Similarity.cs [hunk 2]
```diff
@@ -3,8 +3,8 @@
 namespace Nest
 {
 	/// <summary>
-	/// BM25 Similarity. Introduced in Stephen E. Robertson, Steve Walker, Susan Jones, Micheline Hancock-Beaulieu, 
-	/// and Mike Gatford. Okapi at TREC-3. In Proceedings of the Third Text REtrieval Conference (TREC 1994). Gaithersburg, USA, November 1994.
+	/// BM25 Similarity. Introduced in Stephen E. Robertson, Steve Walker, Susan Jones, Micheline Hancock-Beaulieu,
+	/// and Mike Gatford. Okapi at TREC-3. In Proceedings of the Third Text Retrieval Conference (TREC 1994). Gaithersburg, USA, November 1994.
```
Comment (codebrain) [inline src/Nest/IndexModules/Similarity/BM25Similarity.cs]: :+1:

File: src/Nest/IndexModules/Similarity/DefaultSimilarity.cs (hunk 1 above)
Comment (codebrain) [inline src/Nest/IndexModules/Similarity/DefaultSimilarity.cs]: Binary breaking change.

File: src/Tests/Framework/ManagedElasticsearch/Tasks/InstallationTasks/InstallationTaskBase.cs [hunk 3]
```diff
@@ -35,7 +35,7 @@ protected void ExecuteBinary(string binary, string description, params string[]
 			Console.WriteLine($"Preparing to execute: {description} ...");
 			var timeout = TimeSpan.FromSeconds(420);
 			var handle = new XplatManualResetEvent(false);
-			Task.Run(() =>
+			var task = Task.Run(() =>
```
Comment (codebrain) [inline src/Tests/Framework/ManagedElasticsearch/Tasks/InstallationTasks/InstallationTaskBase.cs]: :+1:

File: src/Tests/IndexModules/Similarity/SimilaritySettings.cs [hunk 4]
```diff
@@ -24,21 +24,26 @@ public class Usage : PromiseUsageTestBase<ISimilarities, SimilaritiesDescriptor,
 				def = new
 				{
 					discount_overlaps = true,
-					type = "default"
+					type = "classic"
```
Comment (codebrain) [inline src/Tests/IndexModules/Similarity/SimilaritySettings.cs]: 👍

Comment (russcam) [reply src/Tests/IndexModules/Similarity/SimilaritySettings.cs]: This should be `default` here in 2.x, not `"classic"`

Comment (Mpdreamz) [reply src/Tests/IndexModules/Similarity/SimilaritySettings.cs]: Yep as per: https://www.elastic.co/guide/en/elasticsearch/reference/2.4/index-modules-similarity.html#default-similarity

**Pull request merged**